### PR TITLE
CPLAT-6965 Automatically ignore common binary packages

### PR DESCRIPTION
**Pull request by todbachman-wf**
## Motivation
There are a number of packages that we typically consume only for their binaries. Consumers of dependency_validator have to manually ignore them. It would be convenient if dependency_validator would just ignore them by default.

## Changes
Defined a whitelist of common binary packages that dependency_validator will ignore by default. Also included a flag to turn this behavior off, should consumers want to do that.

#### Release Notes
Automatically ignore common binary packages, including: `build_runner`, `build_test`, `build_web_compilers`, `built_value_generator` and `dart_style`.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/dependency_validator/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/dependency_validator/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/constants.dart [hunk 1]
```diff
@@ -20,6 +20,15 @@ const String nameKey = 'name';
 /// String key in pubspec.yaml for the transformers map.
 const String transformersKey = 'transformers';
 
+/// Packages that are typically only used for their binaries.
+const List<String> commonBinaryPackages = const [
+  'build_runner',
+  'build_test',
+  'build_web_compilers',
+  'built_value_generator',
```
Comment (robbecker-wf) [inline lib/src/constants.dart]: Glad this is `built_value_generator` and not `build_value_generator` like in the PR description.

Comment (todbachman-wf) [reply lib/src/constants.dart]: Lol. Oops.

File: bin/dependency_validator.dart [hunk 2]
```diff
@@ -53,6 +54,10 @@ final ArgParser argParser = new ArgParser()
     help: 'Comma-delimited list of packages to ignore from validation.',
     splitCommas: true,
   )
+  ..addFlag(ignoreCommonBinariesArg,
```
Comment (corwinsheahan-wf) [inline bin/dependency_validator.dart]: Looks like the readme has the usage for this util, can we update that with this flag?

Comment (todbachman-wf) [reply bin/dependency_validator.dart]: Oops, I missed that. Sure can!

**Comment by seanburke-wf** [comment]
### QA +1
Ran master against w_viewer:
```
$ dependency_validator
Validating dependencies for w_viewer

These packages may be unused, or you may be using executables or assets from these packages:
  * build_runner
  * build_test
  * build_web_compilers
  * coverage
  * dart_style
  * sw_toolbox_dart
```

Ran branch against w_viewer:
```
$ dependency_validator
Validating dependencies for w_viewer

These packages may be unused, or you may be using executables or assets from these packages:
  * coverage
  * sw_toolbox_dart
```

**Comment by todbachman-wf** [comment]
@Workiva/release-management-pp

**Pull request merged**